packages/cactus-plugin-keychain-vault/src/test/typescript/integration/plugin-keychain-vault.test.ts

```diff
@@ -35,79 +61,182 @@
 //         new PluginKeychainVault({
 //           instanceId: "valid-value",
 //           keychainId: null as any,
-//         })
+//         }),
 //     );
 //     t.throws(
 //       () =>
 //         new PluginKeychainVault({
 //           instanceId: "valid-value",
 //           keychainId: "",
-//         })
+//         }),
 //     );
 //     t.end();
 //   });
 
-//   test("get,set,has,delete alters state as expected", async (t: Test) => {
-//     const options: IPluginKeychainVaultOptions = {
-//       instanceId: uuidv4(),
-//       keychainId: uuidv4(),
-//     };
-//     const plugin = new PluginKeychainVault(options);
-//     t.equal(plugin.getKeychainId(), options.keychainId, "Keychain ID set OK");
-//     t.equal(plugin.getInstanceId(), options.instanceId, "Instance ID set OK");
-
-//     const key = uuidv4();
-//     const value = uuidv4();
-
-//     const hasPrior = await plugin.has(key);
-//     t.false(hasPrior, "hasPrior === false OK");
-
-//     await plugin.set(key, value);
-
-//     const hasAfter = await plugin.has(key);
-//     t.true(hasAfter, "hasAfter === true OK");
-
-//     const valueAfter = await plugin.get(key);
-//     t.ok(valueAfter, "valueAfter truthy OK");
-//     t.equal(valueAfter, value, "valueAfter === value OK");
-
-//     await plugin.delete(key);
-
-//     const hasAfterDelete = await plugin.has(key);
-//     t.false(hasAfterDelete, "hasAfterDelete === false OK");
-
-//     const valueAfterDelete = await plugin.get(key);
-//     t.notok(valueAfterDelete, "valueAfterDelete falsy OK");
-
-//     t.end();
-//   });
-
-//   test("rotateEncryptionKeys() fails fast", async (t: Test) => {
-//     const options: IPluginKeychainVaultOptions = {
-//       instanceId: uuidv4(),
-//       keychainId: uuidv4(),
-//     };
-//     const plugin = new PluginKeychainVault(options);
-
-//     const promise = plugin.rotateEncryptionKeys();
-//     const expected = /not implemented/;
-//     await t.rejects(promise, expected, "rotateEncryptionKeys() rejects OK");
-
-//     t.end();
-//   });
-
-//   test("getEncryptionAlgorithm() returns null", (t: Test) => {
-//     const options: IPluginKeychainVaultOptions = {
-//       instanceId: uuidv4(),
-//       keychainId: uuidv4(),
-
-//     };
-//     const plugin = new PluginKeychainVault(options);
-
-//     t.notok(plugin.getEncryptionAlgorithm(), "encryption algorithm falsy OK");
-
-//     t.end();
-//   });
+test("get,set,has,delete alters state as expected", async (t: Test) => {
+  const vaultTestContainer = new VaultTestServer({});
+  await vaultTestContainer.start();
+
+  const ci = await Containers.getById(vaultTestContainer.containerId);
+  const vaultIpAddr = await Containers.getContainerInternalIp(ci);
```

Comment:
@jagpreetsinghsasan Unfortunately this does not work on mac OS, the internal container IPs don't get routed by Docker on mac OS and based on discussions in the reported bugs on Docker for Mac, this is not likely to change in the future either because of some OS level limitation/security feature. This is why you need to use the internal-ip package like we discussed in some other test cases.